Dockerfile

```diff
@@ -0,0 +1,34 @@
+# base image
+FROM python:3.7-buster as base
+
+# poetry set up
+ENV POETRY_HOME=${HOME}/.poetry
+ENV PATH=${POETRY_HOME}/bin:${PATH}
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
+
+WORKDIR /myApp
+COPY poetry.lock pyproject.toml ./
+
+# copy across application code
+COPY ./todo_app ./todo_app
```

Comment:
You have this line after the `RUN poetry install --no-dev` and `RUN poetry install` lines. It means you have to repeat yourself, but it means rebuilds will be faster, because you won't need to rerun poetry install when you only changed files in the todo_app folder.